### PR TITLE
Implement a List of Checkboxes

### PR DESCRIPTION
**Pull request by Bass-03**
I found the need to create a list of checkboxes for task management.
This PR includes a quick way to create a list of all checked or all un-checked boxes, based on MDList class.

Tests are based on MDList's tests, everything passing.

### CODE REVIEW

**Comment by jrg94** [comment]
Great addition! This is definitely something that should be included. I'll take a peek at both pull requests as soon as I can. My initial reaction is that this looks good, but I'm a little hesitant to include extra classes. That said, the idea that a checkbox alone has state (i.e., checked or not) means that we probably can't just add the functionality to MDList. Regardless, I'm very pleased with the documentation and testing!

File: snakemd/generator.py [hunk 1]
```diff
@@ -737,6 +737,38 @@ def verify(self) -> Verification:
...
 
 
+class MDCheckList(MDList):
+    """
+    A markdown CheckBox list has boxes that can be clicked.
+
+    :param Iterable[Union[str, InlineText, Paragraph, MDList]] items: 
+        a "list" of objects to be rendered as a Checkbox list
+    :param bool checked: the state of the checkbox;
+        set to True to render an checked box (i.e., True -> 1. item)
```
Comment (jrg94) [inline snakemd/generator.py:747]:
```suggestion
        set to True to render an checked box (i.e., True -> - [x] item)
```

File: snakemd/generator.py [hunk 2]
```diff
@@ -737,6 +737,38 @@ def verify(self) -> Verification:
...
+    """
+    def __init__(self,  items: Iterable[Union[str, InlineText, Paragraph, MDList]], checked: bool=False) -> None:
+        super().__init__(items, False)
+        self.checked = checked
+    
+    def render(self) -> str:
+        """
+        Renders the markdown Check Box list according to the settings provided.
+        For example, if the the ordered flag is set, an ordered list
```
Comment (jrg94) [inline snakemd/generator.py:756]:
```suggestion
        For example, if the the checked flag is set, a checked list
```

File: tests/test_md_check_list.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        [InlineText("Characters"), inner_list, InlineText("Powers")])
+    assert str(
+        outer_list) == "- [ ] Characters\n  - [ ] Deku\n  - [ ] Bakugo\n  - [ ] Uraraka\n- [ ] Powers"
+
+def test_md_list_nested_checked():
+    inner_list = MDCheckList(
+        [InlineText("Deku"), InlineText("Bakugo"), InlineText("Uraraka")], checked=True)
+    outer_list = MDCheckList(
+        [InlineText("Characters"), inner_list, InlineText("Powers")], checked=True)
+    assert str(outer_list) == "- [X] Characters\n  - [X] Deku\n  - [X] Bakugo\n  - [X] Uraraka\n- [X] Powers"
```
Comment (jrg94) [inline tests/test_md_check_list.py:57]: I feel like there needs to be a test that shows that the checkbox class can be mixed with normal MDList objects.

**Review by jrg94 — CHANGES_REQUESTED** [review]
So far so good! Just some subtle changes and thoughts.

**Comment by jrg94** [comment]
One thing I'm thinking about is how this mixes with your other pull request. Maybe this isn't needed and instead we can have a checkbox InlineText object that you can use when creating an MDList. Then, in the functional portion of the code (under Document) we can have a function that does basically what this class does.

**Comment by Bass-03** [comment]
I did had that idea after creating this one, but I think both approaches have value, like this PR can be a quick way to create a checklist, just quick and fast.

While the other PR gives more granular control, that is why I kept them both.
Thanks for suck a quick review!

**Comment by jrg94** [comment]
That argument makes sense to me! I'll go ahead and merge it in. One I get both pull requests in, I'll release version 0.10.0.

**Pull request merged**